### DISCUSSION TITLE
Depth Map only option

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#173] [General]
is there a depth map only option where i can convert only the depth maps without the rgb files? and if not could you please add a depth map only option that doesnt have rgb files. Still keep the export disparity as i like the option, just asking for an additional Depth Map Only option. The reason i ask is because this would be handy in saving space when someone only needs the depth map and not the rgb files. Something like this would allow me to also see if i could pull out more detail doing a 518 depth map based off an 8k source, i have the space to do the depth map off the 8k source, just not the space for the 8k rgb. Having this depth map only option would help me save space and conversion time since im creating the rgb files in davinci. I have found a clever work around on saving rgb space in davinci. The only way we learn where the limit is , is by pushing it. Sounds extreme but if we don't test it, we don't know whats possible. That bad blood video shows indeed iw3 with a combination of 16 bit depth maps from davinci can create something remarkable.

 Also i managed to achieve some really near perfect results on a 4 minute video , i used the taylor swift bad blood video to test, i generated my own 16 bit png files with davinci and replaced the rgb files with my own, it allowed me to finally get to a level of no banding. The only issue in the video was one section that was oversharpened, and thats because i originally oversharpened the video because i know youtube loses a lot of sharpness due to compression when it uploads, out side of that it had good depth and no mostly no artifacts except for the car scene that looked over sharpened. See video below:
https://www.youtube.com/watch?v=6DVEErnoYQE

Because depth maps degrade after 518 depth resolution. i did a clever work around by upscaling the depth map equivalent to the width size of 2000 depth resolution in davinci resolve and then also replaced my depth maps in the yml folder and used the awesome new function you have where i entered 2000 on depth resolution and i entered the appropriate width in the new width field and it worked. Thats how was able to achieve near artifact free on that video demo above. This took massive amounts of space to do this four minute video, it took 2 terabtyes because i had to regenerate the rgb files again. i could have saved space if it would have been only a depth map option making it easier for me to longer videos. I have cracked the code on what is needed to get near perfect auto 3d, but it requires a mix of 16 bit 8k and high depth resolution to achieve it.

**Comment by nagadomi** [comment]
I have changed Export feature to not output rgb files when an **image folder** is specified.
You can output only depth with the following workflow,

1. Export video frame imsages with davinci resolve or some other software
2. Specify that image folder in iw3 and Export

In the output yml file, `rgb_dir` is set to the **full path** of the specified image folder. 
example:
```yaml
type: images
fps: 1
rgb_dir: /home/nagadomi/tmp/test_videos/images
depth_dir: depth
...
```
With this yml, SBS images(not video) are output when importing. If you want to encode video with other software, you can use this SBS image sequence.
If you want to output in video format, you will need to make the following changes.

1. Change `type` from `images` to `video`
2. Fix to the correct `FPS` 
3. Specify the audio file (optional)
4. Add `filename` field (optional)

example
```yaml
type: video
fps: 30
filename: my_video
audio_file: /home/nagadomi/tmp/test_videos/input_video.mp4
rgb_dir: /home/nagadomi/tmp/test_videos/images
depth_dir: depth
...
```

The reason for recommending this workflow is,
when rgb and depth images are output separately, the frame numbers may not match for various reasons.

----

In addition, the image library (PIL) currently used in iw3 does not support 16-bit RGB.
This does not mean that 16-bit RGB cannot be loaded, 16-bit RGB image is automatically converted to 8-bit RGB when loaded.
So, on iw3 side, there is no advantage to the rgb image is 16-bit RGB format.

**Reply by kaelsonofkrypto** [reply]
> I have changed Export feature to not output rgb files when an **image folder** is specified. You can output only depth with the following workflow,
> 
> 1. Export video frame imsages with davinci resolve or some other software
> 2. Specify that image folder in iw3 and Export
> 
> In the output yml file, `rgb_dir` is set to the **full path** of the specified image folder. example:
> 
> ```yaml
> type: images
> fps: 1
> rgb_dir: /home/nagadomi/tmp/test_videos/images
> depth_dir: depth
> ...
> ```
> 
> With this yml, SBS images(not video) are output when importing. If you want to encode video with other software, you can use this SBS image sequence. If you want to output in video format, you will need to make the following changes.
> 
> 1. Change `type` from `images` to `video`
> 2. Fix to the correct `FPS`
> 3. Specify the audio file (optional)
> 4. Add `filename` field (optional)
> 
> example
> 
> ```yaml
> type: video
> fps: 30
> filename: my_video
> audio_file: /home/nagadomi/tmp/test_videos/input_video.mp4
> rgb_dir: /home/nagadomi/tmp/test_videos/images
> depth_dir: depth
> ...
> ```
> 
> The reason for recommending this workflow is, when rgb and depth images are output separately, the frame numbers may not match for various reasons.
> 
> In addition, the image library (PIL) currently used in iw3 does not support 16-bit RGB. This does not mean that 16-bit RGB cannot be loaded, 16-bit RGB image is automatically converted to 8-bit RGB when loaded. So, on iw3 side, there is no advantage to the rgb image is 16-bit RGB format.

thanks for adding this depth only function, really appreciate it
